be/src/olap/utils.cpp

```diff
@@ -1264,6 +1265,15 @@ bool valid_datetime(const string &value_str) {
     }
 }
 
+bool valid_bool(const std::string& value_str) {
+    if (value_str == "0" || value_str == "1") {
+        return true;
+    }
+    StringParser::ParseResult result;
+    StringParser::string_to_bool(value_str.c_str(), value_str.length(), &result);
```

Comment:
I found string_to_bool will convert "Truetest" to be true, it is ok?
```
543 inline bool StringParser::string_to_bool_internal(const char* s, int len, ParseResult* result) {
544     *result = PARSE_SUCCESS;
545
546     if (len >= 4 && (s[0] == 't' || s[0] == 'T')) {
547         bool match = (s[1] == 'r' || s[1] == 'R') &&
548             (s[2] == 'u' || s[2] == 'U') &&
549             (s[3] == 'e' || s[3] == 'E');
550         if (match && LIKELY(is_all_whitespace(s + 4, len - 4))) {
551             return true;
552         }
553     } else if (len >= 5 && (s[0] == 'f' || s[0] == 'F')) {
554         bool match = (s[1] == 'a' || s[1] == 'A') &&
555             (s[2] == 'l' || s[2] == 'L') &&
556             (s[3] == 's' || s[3] == 'S') &&
557             (s[4] == 'e' || s[4] == 'E');
558         if (match && LIKELY(is_all_whitespace(s + 5, len - 5))){
559             return false;
560         }
561     }
```

Reply:
"Truetest" won't be converted to `true`  because of `is_all_whitespace()`, it will return  `PARSE_FAILURE`. In additional, this is just test the string is `true` or `false` ignore case;

Reply:
If "true " will be ok ? @yangzhg

Reply:
in BoolLiteral.BoolLiteral(String value) funciton, string "1" is  consider to true. but in string_to_bool_internal(), string "1"  is not consider to true, so it is puzzle. maybe you can unify it.